### PR TITLE
Allow injection of ExecutorService

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/89791 by allowing DI for `ExecutorService`.

The unrelated test was having strange issues around resetting FlutterInjector, but doesn't actually need that at all for the way its set up/written, so I just removed that.

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java [hunk 1]
```diff
@@ -155,14 +172,7 @@ public InitResult call() {
...
             // It helps to reduce time cost of engine setup that blocks the platform thread.
-            Executors.newSingleThreadExecutor()
-                .execute(
-                    new Runnable() {
-                      @Override
-                      public void run() {
-                        flutterJNI.prefetchDefaultFontManager();
-                      }
-                    });
+            executorService.execute(() -> flutterJNI.prefetchDefaultFontManager());
```
Comment (linroid) [inline shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java:175]: The default `executorService` is `Executors.newSingleThreadExecutor()` which execute task sequentially, so will this change slow down the engine startup?

File: shell/platform/android/io/flutter/FlutterInjector.java [hunk 2]
```diff
@@ -132,13 +140,22 @@ public Builder setFlutterJNIFactory(@NonNull FlutterJNI.Factory factory) {
...
+      return this;
+    }
+
     private void fillDefaults() {
       if (flutterJniFactory == null) {
         flutterJniFactory = new FlutterJNI.Factory();
       }
 
+      if (executorService == null) {
+        executorService = Executors.newSingleThreadExecutor();
```
[the remark this answers is not in the document]
Comment (blasten) [reply shell/platform/android/io/flutter/FlutterInjector.java:154]: +1

[the remark this answers is not in the document]
Comment (dnfield) [reply shell/platform/android/io/flutter/FlutterInjector.java:154]: Adding this

File: shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java (hunk 1 above)
Comment (dnfield) [reply shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java:175]: Good catch. Changing this to cached thread pool and adding a test.

**Review by blasten — APPROVED** [review]
LGTM

**Pull request merged**